detection_rules/rule.py

```diff
@@ -238,7 +238,12 @@ def build(cls, path=None, rule_type=None, required_only=True, save=True, **kwarg
                     tactic = schema_prompt('mitre tactic name', type='string', enum=TACTICS, required=True)
                     technique_ids = schema_prompt(f'technique IDs for {tactic}', type='array', required=True,
                                                   enum=list(technique_lookup))
-                    threat_map.append(build_threat_map_entry(tactic, *technique_ids))
+
+                    try:
+                        threat_map.append(build_threat_map_entry(tactic, *technique_ids))
+                    except KeyError as e:
```

Comment:
not a huge fan, since KeyError is pretty generic and you don't know for sure where it's coming from.
but I think this will go away with a prompt loop redesign anyway